Lab1/lab1_polynomial.cpp

```diff
@@ -213,13 +213,18 @@ Polynomial Polynomial::operator-(const Polynomial &target) {
     }
     for (int i = 0; i <= subtractedPolys.data.size(); i++) {
         if (i > target.data.size()) {
-            subtractedPolys.data.push_back(this->data[i]);
+            subtractedPolys.data[i] = data[i];
         } else if (i > this->data.size()) {
-            subtractedPolys.data.push_back(0 - target.data[i]);
+            subtractedPolys.data[i] = (0 - target.data[i]);
```

Comment:
can you explain what this line is doing here?

Reply:
yeah, basically if the target is a bigger size than the polynomial it's subtracting from, it will just have the negative of the coefficient for as many of the first coefficients as needed. basically: 
(2x^2 + 1x + 5) - (5x^4 + 3x^3 + 1x^2 + x + 3) 
the x^4 and x^3 terms of the answer are -5x^4, -3x^3, which is the negatives of the coefficient.